pkg/csi/service/osutils/linux_os_utils.go

```diff
@@ -933,19 +933,32 @@ func (osUtils *OsUtils) IsTargetInMounts(ctx context.Context, path string) (bool
 // Defaults to nfs4 for file volume and ext4 for block volume when empty string
 // is observed. This function also ignores default ext4 fstype supplied by
 // external-provisioner when none is specified in the StorageClass
-func (osUtils *OsUtils) GetVolumeCapabilityFsType(ctx context.Context, capability *csi.VolumeCapability) string {
+func (osUtils *OsUtils) GetVolumeCapabilityFsType(ctx context.Context,
+	capability *csi.VolumeCapability) (string, error) {
 	log := logger.GetLogger(ctx)
 	fsType := strings.ToLower(capability.GetMount().GetFsType())
-	log.Debugf("FsType received from Volume Capability: %q", fsType)
+	log.Infof("FsType received from Volume Capability: %q", fsType)
 	isFileVolume := common.IsFileVolumeRequest(ctx, []*csi.VolumeCapability{capability})
-	if isFileVolume && (fsType == "" || fsType == "ext4") {
-		log.Infof("empty string or ext4 fstype observed for file volume. Defaulting to: %s", common.NfsV4FsType)
-		fsType = common.NfsV4FsType
-	} else if !isFileVolume && fsType == "" {
-		log.Infof("empty string fstype observed for block volume. Defaulting to: %s", common.Ext4FsType)
-		fsType = common.Ext4FsType
-	}
-	return fsType
+	if isFileVolume {
+		if fsType == "" || fsType == "ext4" {
+			log.Infof("empty string or ext4 fstype observed for "+
+				"file volume. Defaulting to: %s", common.NfsV4FsType)
+			fsType = common.NfsV4FsType
+		} else {
+			return "", logger.LogNewErrorCodef(log, codes.FailedPrecondition,
+				"unsupported fsType %q observed for file volume", fsType)
+		}
+	} else {
+		if fsType == "" {
+			log.Infof("empty string fstype observed for "+
+				"block volume. Defaulting to: %s", common.Ext4FsType)
+			fsType = common.Ext4FsType
+		} else if fsType == common.NfsFsType || fsType == common.NfsV4FsType || fsType == common.NTFSFsType {
```

Comment:
Should we add `xfs` to this list? Instead of so many conditions, can't we check if fstype != `ext4` or `ext3`?

Reply:
done

```golang
		} else if !(fsType == common.Ext4FsType || fsType == common.Ext3FsType || fsType == common.XFSType) {
			return "", logger.LogNewErrorCodef(log, codes.FailedPrecondition,
				"unsupported fsType %q observed for block volume", fsType)
		}
```